pkg/operator/controller/dns/controller.go

```diff
@@ -573,14 +576,23 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 			log.Info("using fake DNS provider because cluster's ControlPlaneTopology is External")
 			return &dns.FakeProvider{}, nil
 		}
+
 		var err error
-		cisInstanceCRN := platformStatus.IBMCloud.CISInstanceCRN
-		if cisInstanceCRN == "" {
-			return nil, fmt.Errorf("missing cis instance crn")
-		}
-		dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, cisInstanceCRN, userAgent)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create IBM DNS manager: %v", err)
+		if platformStatus.IBMCloud.CISInstanceCRN != "" {
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, platformStatus.IBMCloud.CISInstanceCRN, userAgent, true)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create ibm cis dns provider: %v", err)
+			}
+		} else if platformStatus.IBMCloud.DNSInstanceCRN != "" {
+			rExp := regexp.MustCompile(ibm.IBMResourceCRNRegexp)
+			matches := rExp.FindStringSubmatch(platformStatus.IBMCloud.DNSInstanceCRN)
+			if len(matches) <= 0 {
+				return nil, fmt.Errorf("CRN does not match expected format: %s", platformStatus.IBMCloud.DNSInstanceCRN)
+			}
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, matches[rExp.SubexpIndex("guid")], userAgent, false)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create ibm dnssvcs dns provider: %v", err)
```

Comment:
```suggestion
				return nil, fmt.Errorf("failed to create IBM Cloud DNS Services provider: %w", err)
```
or
```suggestion
				return nil, err
```
per my other comment.

Reply:
Changed it.